### ISSUE TITLE
The end of process takes 10 seconds after the program is closed

### ISSUE THREAD

**Issue by BITONG-Z** [eclipse-paho/paho.mqtt.cpp#379]
When the client connects to the broker, it either closes the application or disconnects first and then closes the application, which process takes about 10 seconds to end!
[eap.txt](https://github.com/eclipse/paho.mqtt.cpp/files/8160388/eap.txt)

**Comment by Pascal-Fremaux** [comment]
Hello, I recommend that you switch to the new Paho C 1.3.10 that fix some problems that might be related to your issue.
I do not know you code, but you have a lot of memory leak and a delayed MQTTAsync_stop (10s timer somewhere?), so review your code as you are doing something unsafe.

**Comment by fpagliughi** [comment]
Yes, I believe that the new Paho C 1.3.10 fixes a number of the small delay issues, particularly at shutdown, though I've never seen anything as long as 10sec. I'm going to start testing the new C lib with this one and update issues accordingly. If you have time to test it, let us know if it changes the behavior.

**Comment by BITONG-Z** [comment]
Thank you very much for your reply. the new Paho C 1.3.10 con't fix my issue ,I'll take a closer review my code. But I don't know if the different operating systems have an impact. The same program that a simple program to test connection works fine on Linux, but shuts down slowly on Windows. Thanks for your help.
[MQTT_TEST_DLL.zip](https://github.com/eclipse/paho.mqtt.cpp/files/8485931/MQTT_TEST_DLL.zip)

**Comment by fpagliughi** [comment]
This has continuously improved in the last few releases of the C lib.

Feel free to re-open if this is still a problem.
